compautogen.c

```diff
@@ -0,0 +1,56 @@
+#include <limits.h>
+#include <stdio.h>
+
+enum { no_argument, required_argument, optional_argument, };
+struct option {
+    const char *name;
+    int has_arg;
+    int *flag;
+    int val;
+    const char *desc;
+};
+#define DESC(X) X
+
+#include "src/opts.h"
+
+int main(void)
+{
+    enum { lopts_len = (sizeof(lopts) / sizeof(lopts[0])) - 1 };
+
+#if 0
+    char short_opts[lopts_len * 3 + 1];
+    size_t n = 0;
+    for (const struct option *o = lopts, *end = o + lopts_len; o < end; ++o) {
+        if (o->val >= UCHAR_MAX)
+            break;
+        short_opts[n++] = o->val;
+        if (o->has_arg == required_argument)
+            short_opts[n++] = ':';
+        else if (o->has_arg == optional_argument)
+            short_opts[n++] = ':', short_opts[n++] = ':';
+    }
+    short_opts[n++] = '\0';
+    puts(short_opts);
+    return 0;
+#endif
```

Comment:
Was a bit annoying having to adjust both `stropts` and `lopts` when adding an option so was playing around to see the viability of generating the short opts string from the lopts array.

Turned out to be bigger than I wanted it to be, so I'll probably drop this idea.